### PR TITLE
Some static analyzer warning suppression

### PR DESCRIPTION
**Pull request by akva2**
- explicit constructors
- clarify operator precedence
- catch polymorphic exceptions by reference

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Comment by akva2** [comment]
jenkins build this with downstreams please

File: src/ASM/ASMs2D.h [hunk 1]
```diff
@@ -65,7 +65,8 @@ class ASMs2D : public ASMstruct, public ASM2D
     int  indxI;    //!< Running node index in the local I-direction
 
     //! \brief Default constructor.
-    BlockNodes() { iinod = inc[0] = inc[1] = 0; indxI = 1; }
+    BlockNodes()
+    { memset(ibnod,0,4*sizeof(int)); iinod = inc[0] = inc[1] = nnodI = 0; indxI = 1; }
```
Comment (kmokstad) [inline src/ASM/ASMs2D.h]: Cosmetic: Use same layout as the corresponding ASMs3D constructor

**Pull request merged**